torch/csrc/utils/future.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <ATen/core/ivalue.h>
+
+namespace torch {
+
+namespace utils {
+
+// FutuerError inherits from std::exception, it can return const char* or
+// std::string error message
+class TORCH_API FutureError final : public std::exception {
+public:
+  FutureError(std::string&& errorMsg) : errorMsg_(std::move(errorMsg)) {}
```

Comment:
You can pass by value and then move it into place. The caller can `std::move` it into the constructor if they wish.